### PR TITLE
Respect exit codes

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/2XWgdfo7/5-%F0%9F%92%A5-make-setup-super-fast-and-easy

This fixes an issue we've been having where the CLI was always exiting with '0', even when the invoked command was malformed or had failed.

### CODE REVIEW

File: lib/commands/base.rb [hunk 1]
```diff
@@ -4,23 +4,26 @@
 
 module Commands
   class Base
+    attr_reader :config_directory, :service, :stack, :verbose
```
Comment (thomasleese) [inline lib/commands/base.rb]: Do these need to be public?

Comment (benthorner) [reply lib/commands/base.rb]: Ah, no - it got a bit mixed up in the rebase. Private again.

**Comment by benthorner** [comment]
@bevanloon do you think we can incorporate this technique into the doctors?

**Comment by bevanloon** [comment]
@benthorner I think at the moment, it would fail the individual check if the system calls returned an error anyway, so people will still realise something is broken (if they read the output). But yeah - we should probably add this in as an enhancement because more info is better when debugging.

**Pull request merged**